chunky/src/java/se/llbit/chunky/block/Block.java

```diff
@@ -97,31 +91,67 @@ public String toString() {
     return name;
   }
 
+  /**
+   * Check if this block is a block entity, i.e. {@link #createBlockEntity(Vector3, CompoundTag)} should be invoked to
+   * create an entity from this block and its entity data tag. This is used for blocks that need to create a new entity
+   * that needs the block entity data, e.g. signs.
+   * <p>
+   * This is mutually exclusive with {@link #hasEntities()}, use that one if you don't need block entity data.
+   *
+   * @return True if this block is a block entity, false otherwise
+   */
   public boolean isBlockEntity() {
     return false;
   }
 
-  public Entity toBlockEntity(Vector3 position, CompoundTag entityTag) {
-    throw new Error("This block type can not be converted to a block entity: "
+  /**
+   * Create a block entity from this block and the given block entity tag at the specified position.
+   *
+   * @param position  Position
+   * @param entityTag Block entity tag
+   * @return The block entity created from this block's data and the entity tag
+   * @throws UnsupportedOperationException If this block is not a block entity (i.e. {@link #isBlockEntity()} returns false
+   */
+  public Entity createBlockEntity(Vector3 position, CompoundTag entityTag) {
```

Comment:
An additional thought, is it possible for a user to need both the entity tag and the ability to return multiple entities?

Reply:
The closest we have is the decorated pot, which uses the block entity information to update the block tag (creating a new `Block` instance) and then uses an entity to create the upper part (which is higher than 1x1x1 and thus can't be in the octree).